ArmRobot/Assets/Scripts/RandomizerPositionObject.cs

```diff
@@ -0,0 +1,166 @@
+            float randomZ = 0.25f;
+            int randomSignX = 2 * Random.Range(0,2) - 1;
+            if (randomSignX == 1){
+                randomX = minRadius + gameObjectRadius + (maxRadius - minRadius - 2 * gameObjectRadius) * Random.value;
+            }
+            else {
+                randomX = -maxRadius + gameObjectRadius + (maxRadius - minRadius - 2 * gameObjectRadius) * Random.value;
+            }
+
+            int randomSignZ = 2 * Random.Range(0,2) - 1;
+            if (randomSignZ == 1){
+                randomZ = minRadius + gameObjectRadius + (maxRadius - minRadius - 2 * gameObjectRadius) * Random.value;
+            }
+            else {
+                randomZ = -maxRadius + gameObjectRadius + (maxRadius - minRadius - 2 * gameObjectRadius) * Random.value;
+            }
+            Vector2 randomPoint = new Vector2(randomX, randomZ); // now we are sure the point is in the area reachable by the robot 
+
+            // now we need to check if their is no conflict with the position of objects already moved 
+            // we call the PositionIsValid ;ethod and check whether or not the random point generated is a good fit
+            bool goodPoint = PositionIsValid(randomPoint, listOfAlreadyMovedObjects);
+
+            if (goodPoint == true){
+                return randomPoint;
+            }
+        }
+    }
+
+    bool PositionIsValid(Vector2 randomPoint, List<GameObject> listOfAlreadyMovedObjects){
+        /* method to check if their is no conflict with the position of objects already moved 
+        and the point we have generated. 
+        */
+
+        bool goodPoint = true; 
+            
+        // we iterate through all the list of already moved objects 
+        for (var i = 0; i < listOfAlreadyMovedObjects.Count; i++){
+            gameObjectSeen = listOfAlreadyMovedObjects[i];
+            // first we will compute the distance between the datapoint generated and one of the object of the list of already moved objects 
+            Vector2 interObject = new Vector2(gameObjectSeen.transform.position.x - randomPoint[0], gameObjectSeen.transform.position.z - randomPoint[1]);
+            float distanceBetweenObjects = interObject.magnitude;
+            
+            // now we need to access to the radius of the gameobjectSeen
+            float gameObjectSeenRadius = GameObjectRadius(gameObjectSeen);
+            
+            // now we need to access to the radius of the current object
+            float gameObjectRadius = GameObjectRadius(gameObject);
+
+            /* keep only if the center of the object from the list of already moved objects is superior 
+            * to the sum of their respective radius  
+            */
+            if (distanceBetweenObjects <= gameObjectRadius + gameObjectSeenRadius){
+                goodPoint = false;
+                break;
+            }
+        }
+        return goodPoint;
+    }
+
+    public float GameObjectRadius(GameObject gameObject){
```

Comment:
If you make the change I suggested in the comment below, then this method would rely on only its arguments as inputs - which means you could make it static (`public static float GameObjectRadius...`). It is good to make methods static when you can.